internal/stack/resources.go

```diff
@@ -171,6 +184,45 @@ func applyResources(profile *profile.Profile, stackVersion string) error {
 	return nil
 }
 
+func addClientCertsToResources(resourceManager *resource.Manager, certResources []resource.Resource) error {
+	certPath := filepath.Join(CertsFolder, ElasticAgentFolder, "cert.pem")
+	keyPath := filepath.Join(CertsFolder, ElasticAgentFolder, "key.pem")
+
+	var certFile, keyFile string
+	var err error
+	for _, r := range certResources {
+		res, _ := r.(*resource.File)
+
+		if strings.Contains(res.Path, ElasticAgentFolder) {
+			var buf bytes.Buffer
+			if res.Path == certPath {
+				err = res.Content(nil, &buf)
+				if err != nil {
+					return fmt.Errorf("failed to read client certificate: %w", err)
+				}
+				// Replace newlines with spaces to create proper indentation in the config
+				certFile = buf.String()
+				continue
+			}
+			if res.Path == keyPath {
+				err = res.Content(nil, &buf)
+				if err != nil {
+					return fmt.Errorf("failed to read client key: %w", err)
+				}
+				// Replace newlines with spaces to create proper indentation in the config
+				keyFile = buf.String()
+				continue
+			}
+		}
+	}
+
+	resourceManager.AddFacter(resource.StaticFacter{
+		"logstash_ssl_certificate": certFile,
+		"logstash_ssl_key":         keyFile,
```

Comment:
Nit. Should we call it agent or client certificates? As they are not the logstash certificates.
```suggestion
		"agent_certificate": certFile,
		"agent_key":         keyFile,
```
```suggestion
		"logstash_client_certificate": certFile,
		"logstash_client_key":         keyFile,
```